docs/src/test-runners-java.md

```diff
@@ -186,3 +186,75 @@ junit.jupiter.execution.parallel.mode.classes.default = concurrent
 junit.jupiter.execution.parallel.config.strategy=dynamic
 junit.jupiter.execution.parallel.config.dynamic.factor=0.5
 ```
+
+## TestNg
+
+In TestNg you can initialize [Playwright] and [Browser] in [@BeforeSuite](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/BeforeAll.html) method and destroy them in [@AfterSuite](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/AfterAll.html). In the example below all three test methods use the same [Browser]. Each test uses its own [BrowserContext] and [Page].
+
+```java
+package org.example;
+
+import com.microsoft.playwright.Browser;
+import com.microsoft.playwright.BrowserContext;
+import com.microsoft.playwright.Page;
+import com.microsoft.playwright.Playwright;
+import org.testng.annotations.*;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestExample {
+  // Shared between all tests in this class.
+  static Playwright playwright;
+  static Browser browser;
+
+  // New instance for each test method.
+  BrowserContext context;
+  Page page;
+
+  @BeforeSuite
+  static void launchBrowser() {
+    playwright = Playwright.create();
+    browser = playwright.chromium().launch();
+  }
+
+  @AfterSuite
+  static void closeBrowser() {
+    playwright.close();
+  }
+
+  @BeforeMethod
+  void createContextAndPage() {
+    context = browser.newContext();
+    page = context.newPage();
+  }
+
+  @AfterMethod
+  void closeContext() {
```

Comment:
Please make sure that the context and page fields are reset to null so that in case of failure in after `@BeforeMethod` next time we don't end up closing the same context again. Something similar [to this](https://github.com/microsoft/playwright-java/blob/aef9badd64f70b8ae7369a212a0a246dfe5d7951/playwright/src/test/java/com/microsoft/playwright/TestBase.java#L141-L147).

Reply:
@yury-s  i will test the testng with null context and page to null and update you

Reply:
Friendly ping @bhargavkumar-65 are you planning to work on the requested changes?

Reply:
> Friendly ping @bhargavkumar-65 are you planning to work on the requested changes?

Yes, Please allow me couple of days to finish this